### DISCUSSION TITLE
How to provide a base response to /trpc route

### DISCUSSION THREAD

**Discussion by michealroberts** [trpc/trpc#3858] [Q&A]
I was wondering if someone could provide me an example of how to respond from the /trpc base route with a generic { "hello": "world!" } for version 10. I can't seem to specifically reference anything in the docs for v10.

I was perhaps thinking something like this would work:

```ts
export const appRouter = router({
  '': publicProcedure.query(() => {
    return {
      message: 'Welcome to the Observerly API'
    }
  }),
  ...otherRoutesHere
})
```

**Comment by KATT** [comment]
I would expect the above to work but also feels like it could mess up the inference of how the trpc client works. What do you need this for?

**Comment by michealroberts** [comment]
Yeh me too, but I am seeing a 404. I have also tried with a pointer of '/' ...

In version 9, I used this to success:

```ts
export const appRouter = createRouter()
  .query('', {
    resolve() {
      return {
        description: ".",
        endpoint: '/trpc',
        name: ''
      }
    }
  }),
  ...
```

The other routes could be accessed without issue (both get, post and subscribe procedures).

Good question, I basically used it to dynamically extract and pull in all of the available public route 🙏  and to redirect from a 404 page to "something" if there isn't a hit. I treated it like a base URL for my API ...

**Comment by michealroberts** [comment]
Just to add additional context, the above provided for v10 does resolve when I navigate explicitly to the following: http://localhost:3010/api/v1/ (i.e., with the SLASH appended). No slash = no route :( 

🙏
